mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/MailboxListenerRegistry.java

```diff
@@ -23,44 +23,77 @@
 
 import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Supplier;
 
 import com.github.steveash.guavate.Guavate;
 import com.google.common.collect.ImmutableSet;
 import reactor.core.publisher.Flux;
 
 class MailboxListenerRegistry {
+
+    interface RemovalStatus {
+        boolean lastListenerRemoved();
+    }
+
+    public static class Registration {
```

Comment:
It looks confusing to me that we use the exact same wording than the EventBus API `Registration`.

What I do propose is to call this class `LocalRegistration`

Reply:
I agree it is getting messy in term of naming. I think we should rename MailboxListenerRegistry in the first place.